From 6da453e73fe7eabc545a77d94ccdba180808cdae Mon Sep 17 00:00:00 2001
From: Hien Dang <hien.dang.eb@renesas.com>
Date: Wed, 17 Aug 2016 15:12:59 +0700
Subject: [PATCH 176/909] serial: sh-sci: Use pm_runtime_get/put when suspend

commit 3137848bd7f445455f68b98dcec21f26111a817e from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

Due to the commit '39dd0f234fc37d ("PM / Domains: Allow runtime PM during
system PM phases")', runtime PM status might be suspended when back up
module registers. Therefore, it is necessary to ensure device is
on during suspend by using pm_runtime_get/put.

Signed-off-by: Hien Dang <hien.dang.eb@renesas.com>
Signed-off-by: Hiromitsu Yamasaki <hiromitsu.yamasaki.ym@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/tty/serial/sh-sci.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/tty/serial/sh-sci.c b/drivers/tty/serial/sh-sci.c
index 60a2c4211276..e7f9cca632a4 100644
--- a/drivers/tty/serial/sh-sci.c
+++ b/drivers/tty/serial/sh-sci.c
@@ -3265,8 +3265,11 @@ static __maybe_unused int sci_suspend(struct device *dev)
 {
 	struct sci_port *sport = dev_get_drvdata(dev);
 
-	if (sport)
+	if (sport) {
+		pm_runtime_get_sync(sport->port.dev);
 		uart_suspend_port(&sci_uart_driver, &sport->port);
+		pm_runtime_put(sport->port.dev);
+	}
 
 	return 0;
 }
-- 
2.17.1

